_posts/2022-03-13-software.md

```diff
@@ -0,0 +1,60 @@
+When releasing a software it is important to think about under which license you want to release it under. There are many licenses, but they can be grouped into [three groups](https://ieeexplore.ieee.org/document/5370763):
+
+- Free-for-all licenses (permissive) This type of licenses require to acknowledge the original author(s) of the licensed work in any derivative work. However, it allows derivative works to be then published under a different type of license, and/or made proprietary.
+
+- Keep-open licenses This type of licenses require for the derivative work to be also released under open license. However, the derivative work (e.g. a software product) could then be included as one part of a larger proprietary work.
+
+- Share-alike (copyleft) licenses This type of licenses require that, if the licensed work is included in larger work or project, the entire work should be released open-source.
+
+For more information on open source licenses you can find more details [on Open Source initiative](https://opensource.org/licenses).
+
+### [Metadata](https://climatecompatiblegrowth.github.io/guidelines/)
+Metadata is "data about the data" meaning that it describes the context, quality and condition of a e.g. dataset.
+ 
+### [Attribution](https://climatecompatiblegrowth.github.io/guidelines/practice/2022/02/01/attribution.html)
+To make sure that the right people who has contributed to the software development are properly attributed one good practice is that the authors for each script can be added on the top of the file. If documentation software such as [Sphinx](https://www.sphinx-doc.org/en/master/index.html) is used then, if the metadata is added on the top of the file in a certain format, it will be added to the documentation file.
+If the code is collaborativley developed on Github then each author that commits to the main branch will be added as a contributor to the code. If the code is [released](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository) the contributors to the Github repository will be added as contributors/authors to that release.
+
+### [Release](https://climatecompatiblegrowth.github.io/guidelines/)
+A software release can be managed through for example [Github relases](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository). On Github you first make a tag to mark that this version of the code is a specific version. If you also want to make a release of the code you can then create a release based on that tag.
+
+If you make a release on GitHub then this can be linked to your Zenodo account which will give the software a DOI and also make it citable for each release you make.
+  
+## Example of software release - NISMOD
+  
+In the CCG project a Python package has been released for NISMOD [snail](https://github.com/nismod/snail/releases) under different versions using [Python Package Index (PyPi)](https://pypi.org/).
```

Comment:
We might want to think how to improve this example.  Perhaps the PyPSA example would be more descriptive as it works across several of the practices and outputs.

Reply:
I have the PyPSA example in Workflows, but I can add it here as well if you want.